node/network/bridge/src/network.rs

```diff
@@ -157,17 +157,9 @@ impl Network for Arc<NetworkService<Block, Hash>> {
 				let mut found_peer_id = None;
 				// Note: `get_addresses_by_authority_id` searched in a cache, and it thus expected
 				// to be very quick.
-				for addr in authority_discovery
-					.get_addresses_by_authority_id(authority)
-					.await
-					.into_iter()
-					.flat_map(|list| list.into_iter())
-				{
-					let (peer_id, addr) = match parse_addr(addr) {
-						Ok(v) => v,
-						Err(_) => continue,
-					};
-					NetworkService::add_known_address(&*self, peer_id.clone(), addr);
+				let addrs = get_addrs_by_authority_id(authority_discovery, authority).await;
+				for (peer_id, multiaddr) in addrs {
+					NetworkService::add_known_address(&*self, peer_id.clone(), multiaddr);
 					found_peer_id = Some(peer_id);
 				}
 				found_peer_id
```

Comment:
hmm, looking at this code again, req/resp still assumes one peer_id per authorityid?